### PR TITLE
Added new argument to write json from mfile, added execution of argument to post processing

### PR DESCRIPTION
**Pull request by ym1906**
It is useful to have PROCESS dump the mfile to json for UQ work. I have added an argument to do this which is executed in postprocessing

### CODE REVIEW

File: process/main.py [hunk 1]
```diff
@@ -209,14 +216,20 @@ def post_process(self):
         # run, for example.
         if self.args.plot:
             # Check mfile exists, then plot
-            mfile = Path(self.args.mfile)
+            mfile_path = Path(self.args.mfile)
             mfile_str = str(mfile.resolve())
```
Comment (timothy-nunn) [inline process/main.py:220]:
```suggestion
            mfile_str = str(mfile_path.resolve())
```

File: process/main.py [hunk 2]
```diff
@@ -184,6 +185,12 @@ def parse_args(self, args):
             default="MFILE.DAT",
             help="mfile for post-processing/plotting",
         )
+        parser.add_argument(
+            "-mj",
+            "--mfilejson",
+            action="store_true",
+            help="produce a json containing mfile data",
```
Comment (timothy-nunn) [inline process/main.py:192]: Update the help message slightly. Where does it write the JSON file, and with what name?

**Review by timothy-nunn — CHANGES_REQUESTED** [review]
Please could you add an integration test for this (maybe with a once through input file?):
- Does running with this command produce a JSON file at the expected location with the expected name?
- Does the JSON file contain valid JSON?
- Does the JSON contain some outputs we would expect?

File: process/main.py (hunk 2 above)
Comment (ym1906) [reply process/main.py]: parser.add_argument(
            "-mj",
            "--mfilejson",
            action="store_true",
            help="Produce a filled json from --mfile arg in working dir",
        )
Modified like this

File: process/main.py (hunk 1 above)
Comment (ym1906) [reply process/main.py]: if self.args.plot:
            # Check mfile exists, then plot
            mfile_path = Path(self.args.mfile)
            mfile_str = str(mfile_path.resolve())

**Comment by ym1906** [comment]
I addressed the changes, let me know if further improvements are needed.

**Review by timothy-nunn — APPROVED** [review]

**Pull request merged**